corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -37,23 +49,60 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
         ui: {
             submitButton: '#query-submit-button',
+            valueDropdown: 'select.query-field',
         },
 
         events: {
+            'change @ui.valueDropdown': 'changeDropdown',
             'click @ui.submitButton': 'submitAction',
         },
 
-        submitAction: function (e) {
-            e.preventDefault();
-            var payload = {};
-            var fields = $(".query-field");
-            var model = this.parentModel;
-            fields.each(function (index) {
+        getAnswers: function () {
+            var $fields = $(".query-field"),
+                answers = {},
+                model = this.parentModel;
+            $fields.each(function (index) {
                 if (this.value !== '') {
-                    payload[model[index].get('id')] = this.value;
+                    answers[model[index].get('id')] = this.value;
                 }
             });
-            FormplayerFrontend.trigger("menu:query", payload);
+            return answers;
+        },
+
+        changeDropdown: function (e) {
+            e.preventDefault();
+            var $fields = $(".query-field");
+
+            // If there aren't at least two dropdowns, there are no dependencies
+            if ($fields.filter("select").length < 2) {
+                return;
+            }
+
+            var Util = hqImport("cloudcare/js/formplayer/utils/util");
+            var urlObject = Util.currentUrlToObject();
+            urlObject.setQuery(this.getAnswers());
+            urlObject.setDoQuery(false);
+            var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
+            $.when(fetchingPrompts).done(function (response) {
+                for (var i = 0; i < response.models.length; i++) {
+                    var choices = response.models[i].get('itemsetChoices');
+                    if (choices) {
+                        var $field = $($fields.get(i)),
+                            value = $field.val();
+                        $field.html('');
```

Comment:
Can't the HTML be controlled via knockout bindings?

Reply:
This part of web apps doesn't use knockout (knockout is only used within forms), Marionette provides similar functionality. I'll take another look at using Marionette, this code is pretty messy.

Reply:
Updated in 4a2a8b3 to use Marionette better. It still does a bit of jQuery HTML manipulation, but I think that's inevitable, because Backbone/Marionette doesn't support two-way data binding.